File: 2024/01/p2.test.lua
local describe = require("lib.test")
local p2 = require(".2024.01.p2")

describe("input", function(it)
  it("example", function ()
    local example = [[3   4
4   3
2   5
1   3
3   9
3   3]]
    local expected = 31

    local n = p2.solve(example)

    assert(n == expected, string.format("Expected %d, but was %d", expected, n))
  end)

  it("input", function ()
    local fp = io.open("./2024/01/assets/input.txt", "r") or error("Input not found")
    local input = fp:read("*a")
    local expected = 19457120

    local n = p2.solve(input)

    assert(n == expected, string.format("Expected %d, but was %d", expected, n))
  end)
end)

File: 2024/12/__tests__/p2.test.ts
import assert from "node:assert";
import { readFile } from "node:fs/promises";
import { describe, it } from "node:test";
import { solve } from "../p2.js";
import path from "node:path";

describe("p1", () => {
  it("example 1", () => {
const example = `AAAA
BBCD
BBCC
EEEC`;

    assert.strictEqual(solve(example), 80);
  });

  it("example 2", () => {
const example = `OOOOO
OXOXO
OOOOO
OXOXO
OOOOO`;

    assert.strictEqual(solve(example), 436);
  });

  it("example 3", () => {
const example = `EEEEE
EXXXX
EEEEE
EXXXX
EEEEE`;

    assert.strictEqual(solve(example), 236);
  });

  it("input", async () => {
    const filename = path.join(import.meta.dirname, "../assets/input.txt").replace("/dist/", "/");
    const input = await readFile(filename, "utf8");

    assert.strictEqual(solve(input), 921636);
  });
});
